silhouette/app/com/mohiva/play/silhouette/impl/providers/OAuth2Provider.scala

```diff
@@ -137,6 +137,43 @@ trait OAuth2Provider extends SocialProvider with OAuth2Constants with Logger {
     }
   }
 
+  override def authenticate[B](userStateParam: Option[Map[String, String]])(implicit request: ExtractableRequest[B]): Future[Either[StatefulResult, OAuth2Info]] = {
```

Comment:
Why `Option`? If a user uses this method then he probably want's to send state. Also, the `Map` itself can be empty.

I think `Map[String, String]` is enough.